### ISSUE TITLE
Provide a way to check if an actor is ready

### ISSUE THREAD

**Issue by surma** [PolymerLabs/actor-helpers#28]
After trying to work with a pub/sub actor, it became clear that we _need_ to provide a mechanism to check if an actor is ready. If I send a message to `PubSubActor`, and `PubSubActor` hasn’t called `hookup()` yet (because it the worker it lives in is still booting up), the queued-up messages will be deleted. Building a mechanism on top is possible, but proves to be incredibly painful for developers.

I see two options:

1. Make `lookup()` async again and make it resolve when the looked-up actor is ready
2. Provide a new method `waitFor(name: ValidActorName)`

We still have the (currently unused!) channel definitions for `actor.lookup` and `actor.lookup.exists`, so we could use those under the hood.

WDYT?

(Mostly made this issue so I wouldn’t forget the details until Monday :D)

**Comment by rgfoley** [comment]
I noticed the feature exists in the `waitFor` branch. Is there any plan on merging it into master? I was thinking of using it to implement something like "actor dependencies", if that makes sense. So that an actor can ensure that the actors it relies on are spun up. Not entirely sure if that is contradictory to the Actor model as a whole, but I believe I have a good use-case for it.

**Comment by surma** [comment]
No, currently no plans to merge it to master. I needed it back then because we had a different message bus implementation. Our current implementation buffers messages until the actor is ready to process them, solving 95% of the use-cases of `waitFor()`.

If you still find yourself needing a way to check if an actor is ready, you could implement a simple ping/pong message call with [the request/response module](https://github.com/PolymerLabs/actor-boilerplate/blob/example/todo/src/utils/request-response.ts) I built for the TODO demo.

**Comment by rgfoley** [comment]
Okay, I'll check that out. Thanks!

**Comment by rgfoley** [comment]
Perhaps the request/response model is more robust for some use cases, but I think I was able to achieve my goal by creating another BroadcastChannel used to coordinate information about actor readiness. The changes are in my [actor-helpers](https://github.com/rgfoley/actor-helpers/) fork.

I had a somewhat similar `waitFor` implementation, except I also used this channel to keep the readiness in sync between workers since the ready state is otherwise not shared between workers. I'm thinking some kind of meta Actor might be able to get the job done as well.

Thanks again for the awesome code!
